ideafast/middleware-services
# Byteflies update, closes #27

These changes ensure that the ByteFlies branch follows a similar flow to the Dreem branch, including folder structure, batch-processing by patient+device_id as well as logging. 

Partially responds to #28 (at least for testing) 

## Test
- pull down branch
- run `poetry install`
- run `poetry shell`
- run `poetry run nox -rs` for all tests
- run `poetry run consumer` to run the inventory for the overall pipeline
- ensure that you have the mongo DB running
- ensure your `.envs` are correctly set - including `ACCESS_AUTHENTICATED_ENDPOINTS=True` in `.consumer.env`
- run `python data_transfer/main.py BTF Kiel 0 1` to run the BTF pipeline for today (`0`) and yesterday (`1`)

For testing the historical data pipeline, I suggest to add `return` [after logging the data period here](https://github.com/ideafast/middleware-services/blob/352459b483275c166011d71ff435bf17a1aeb2bb/data_transfer/dags/btf.py#L49), and run:
- `python data_transfer/main.py BTF Kiel -1`
